### PR TITLE
Fix for 20 second pause on some systems.

### PR DESCRIPTION
**Pull request by mudlord**
https://github.com/libretro/mupen64plus-libretro-nx/issues/40

Capture

This is a fix for the current 20 or so second loading pause on startup.

The culprit is the usage of srand() and rand(). Not using these functions and using a simple LSFR based number generator makes a massive performance increase and almost eliminates the pause. Its recommended that the function is inlined.

### CODE REVIEW

**Comment by m4xw** [comment]
Thanks for doing this, long time outstanding issue!
So the issue was not enough entropy?
Will review it after the conflict is resolved!
Also don't mind these trolls, I personally don't give a f what they say ;)

**Comment by mudlord** [comment]
Issue with doing on GPU was not enough entropy without using integer bitmath and SM4 features.

Using a simple LSFR on CPU is enough to give decent results.

File: GLideN64/src/NoiseTexture.cpp [hunk 1]
```diff
@@ -34,27 +34,24 @@ NoiseTexture::NoiseTexture()
...
-u32 Rand(u32 rand_value)
+inline u32 irand()
 {
-#ifdef MINGW
-	rand_s(&rand_value);
-#else
-	rand_value = rand();
-#endif
-	return rand_value;
+        static u32 iseed = 1;
```
Comment (m4xw) [inline GLideN64/src/NoiseTexture.cpp]: Mind fixing the space?

Comment (mudlord) [reply GLideN64/src/NoiseTexture.cpp]: fixed?

Comment (m4xw) [reply GLideN64/src/NoiseTexture.cpp]: ye

**Comment by m4xw** [comment]
@mudlord If you don't mind I will hold off merging this PR until I tried it on Switch this evening.
Want to squash it yourself or want me to do it on merge?
Also can u provide a comparison screenshot for reference? Will be like 7h until I can try myself.

**Comment by mudlord** [comment]
do it on merge, and okay.

**Comment by mudlord** [comment]
srand
lsfr

Results are almost identical.

**Comment by m4xw** [comment]
Nice, thanks!

**Comment by twinaphex** [comment]
Outstanding work. This really stuck out like a sore thumb on Windows.

**Comment by mudlord** [comment]
No doubt, I was just extremely surprised why no one bothered to debug thoroughly **why** things were so slow. 

Didn't make sense to use a cryptographically secure RNG which uses thread local storage (on Windows at least, don't know how the CRT implements under Mac/Linux kernel), when a super simple LSFR based gen would suffice.

**Comment by m4xw** [comment]
> why no one bothered to debug thoroughly why things were so slow.

Got other things to do + I only use Switch actively where this Issue doesn't exist, since we feed extra entropy there :P
If u look to blame someone, blame my laziness ;)

**Comment by mudlord** [comment]
that and prob the Switch does things differently when it comes to RNG.

**Comment by m4xw** [comment]
Well it provides hwrng, however its only used to seed newlib sufficiently iirc.

**Comment by mudlord** [comment]
hwrng? nice.....is it the shit show that intel's hwrng was?

**Comment by m4xw** [comment]
Its a decent API by the system.

**Comment by m4xw** [comment]
Decided to merge it anyway already.

**Pull request merged**